### PR TITLE
Add golangci-lint config and CI step

### PR DESCRIPTION
**Pull request by Frostman**

### CODE REVIEW

File: hack/tools.mk [hunk 1]
```diff
@@ -149,4 +150,9 @@ $(GCOV2LCOV): $(LOCALBIN)
...
 addlicense: $(ADDLICENSE) ## Download addlicense locally if necessary.
 $(ADDLICENSE): $(LOCALBIN)
-	test -s $(LOCALBIN)/addlicense || GOBIN=$(LOCALBIN) go install github.com/google/addlicense@latest
\ No newline at end of file
+	test -s $(LOCALBIN)/addlicense || GOBIN=$(LOCALBIN) go install github.com/google/addlicense@latest
+
+.PHONY: golangci-lint
+golangci-lint: $(GOLANGCI_LINT) ## Download golangci-lint locally if necessary.
+$(GOLANGCI_LINT): $(LOCALBIN)
+	test -s $(LOCALBIN)/golangci-lint || GOBIN=$(LOCALBIN) go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
```
Comment (vjardin) [inline hack/tools.mk:158]: the file is missing an EOF.

**Review by vjardin — APPROVED** [review]
see my minor comments.

Regarding the license headers, spdx seems to be a better option (cosmetic)

**Comment by Frostman** [comment]
> see my minor comments.
> 
> Regarding the license headers, spdx seems to be a better option (cosmetic)

could you please point me to the spdx tool or doc about lic headers? thx!

**Comment by vjardin** [comment]
> could you please point me to the spdx tool or doc about lic headers? thx!
General context: https://spdx.dev/learn/handling-license-info/
For https://en.wikipedia.org/wiki/Apache_License / for Apache 2.0:
```
Apache-2.0
```
Example of diff:
  https://www.spinics.net/lists/linux-wpan/msg08256.html
is it the question ?

**Comment by Frostman** [comment]
@vjardin Oh, I got it now. I've misread your original comment

**Pull request merged**